doc/how_to_guides/parallel_planning/src/parallel_planning_main.cpp

```diff
@@ -0,0 +1,300 @@
+        return false;
+      }
+
+      if (planning_scene_storage->hasPlanningScene(scene_name))  // Just the world (no robot)
+      {
+        moveit_msgs::msg::PlanningSceneWorld world_meta_data;
+        if (!planning_scene_storage->getPlanningSceneWorld(world_meta_data, scene_name))
+        {
+          RCLCPP_ERROR(LOGGER, "Failed to load planning scene world '%s'", scene_name.c_str());
+          return false;
+        }
+        scene_msg.world = world_meta_data;
+        scene_msg.robot_model_name =
+            "NO ROBOT INFORMATION. ONLY WORLD GEOMETRY";  // this will be fixed when running benchmark
+      }
+      else
+      {
+        RCLCPP_ERROR(LOGGER, "Failed to find planning scene '%s'", scene_name.c_str());
+        return false;
+      }
+    }
+    catch (std::exception& ex)
+    {
+      RCLCPP_ERROR(LOGGER, "Error loading planning scene: %s", ex.what());
+      return false;
+    }
+
+    // Add object to planning scene
+    {  // Lock PlanningScene
+      planning_scene_monitor::LockedPlanningSceneRW scene(moveit_cpp_->getPlanningSceneMonitor());
+      scene->processPlanningSceneWorldMsg(scene_msg.world);
+    }  // Unlock PlanningScene
+
+    RCLCPP_INFO(LOGGER, "Loaded planning scene successfully");
+
+    // Get planning scene query
+    moveit_warehouse::MotionPlanRequestWithMetadata planning_query;
+    std::string query_name = scene_name + "_query";
+    try
+    {
+      planning_scene_storage->getPlanningQuery(planning_query, scene_name, query_name);
+    }
+    catch (std::exception& ex)
+    {
+      RCLCPP_ERROR(LOGGER, "Error loading motion planning query '%s': %s", query_name.c_str(), ex.what());
+    }
+
+    planning_query_request_ = static_cast<moveit_msgs::msg::MotionPlanRequest>(*planning_query);
+    visual_tools_.prompt("Press 'next' in the RvizVisualToolsGui window to start the demo");
+    visual_tools_.trigger();
+    return true;
+  }
+
+  /// \brief Set joint goal state for next planning attempt
+  /// \param [in] panda_jointN Goal value for the Nth joint [rad]
+  void setJointGoal(double const panda_joint1, double const panda_joint2, double const panda_joint3,
+                    double const panda_joint4, double const panda_joint5, double const panda_joint6,
+                    double const panda_joint7)
+  {
+    auto robot_goal_state = planning_component_->getStartState();
+    robot_goal_state->setJointPositions("panda_joint1", &panda_joint1);
+    robot_goal_state->setJointPositions("panda_joint2", &panda_joint2);
+    robot_goal_state->setJointPositions("panda_joint3", &panda_joint3);
+    robot_goal_state->setJointPositions("panda_joint4", &panda_joint4);
+    robot_goal_state->setJointPositions("panda_joint5", &panda_joint5);
+    robot_goal_state->setJointPositions("panda_joint6", &panda_joint6);
+    robot_goal_state->setJointPositions("panda_joint7", &panda_joint7);
+
+    // Set goal state
+    planning_component_->setGoal(*robot_goal_state);
+  }
```

Comment:
... part of me wants to recommend that this accept a `std::vector` or something, but it's fine if you don't want to do that extra work. 

If you went that way you'd have to loop through like this (UNTESTED)

```cpp
  void setJointGoal(std::vector<double> const& panda_joint_angles)
  {
    size_t idx = 0;
    auto robot_goal_state = planning_component_->getStartState();
    for (const auto& joint_angle : panda_joint_angles) { 
        robot_goal_state->setJointPositions("panda_joint" + std::to_string(idx++), joint_angle);
    }
    planning_component_->setGoal(*robot_goal_state);
  }
```

Reply:
I see you point. Maybe this would be worth adding to the PlanningComponent class